### PR TITLE
Jg sexcheck

### PR DESCRIPTION
**Pull request by jigold**
1. I ended up removing the test with the VCF from 1000 genomes

2. I couldn't figure out a good way to get the number of variants used in the computation. It's either in the annotations as "sa.imputegender.T" or we'd have to do RDD.count() 

3. I'm open to naming suggestions if you think something else is better. I didn't want to use "sex check" because that implies comparing to the reported gender and reporting errors -- not what I have implemented here.

### CODE REVIEW

File: src/main/scala/org/broadinstitute/hail/methods/ImputeSexPlink.scala [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  import ImputeSexPlink._
+
+  private def xChrVds =  {
+    if (excludePAR)
+      vds.filterVariants(
+      (v: Variant, va: Annotation, gs: Iterable[Genotype]) => !v.inParX && (v.contig == "X" || v.contig == "23")
+    )
+    else
+      vds.filterVariants(
+        (v: Variant, va: Annotation, gs: Iterable[Genotype]) => v.contig == "X" || v.contig == "23"
```
Comment (cseed) [inline src/main/scala/org/broadinstitute/hail/methods/ImputeSexPlink.scala]: I'd write this with one call to filterVariants: `(v.contig == "X" || v.contig == "23") && (!excludePAR || !v.inParX)`

**Comment by cseed** [comment]
@jigold What's the status of this?  It looks like the serializer zero array value stuff from VSM is still in there.

**Comment by tpoterba** [comment]
Shall I make a PR against this with the implementation I envisioned?

**Comment by jigold** [comment]
@tpoterba : That's fine with me if you want to try your approach.

**Comment by tpoterba** [comment]
I still need to change g.gt to g.nNonRefAlleles, I'll push that to this branch (it's a small change).  Is that the right thing to do?  Not really, because you could have a het non ref.  This is actually non-trivial.

**Comment by jigold** [comment]
Should we make this so it requires splitMulti to have been run? Otherwise, is the MAF we're currently calculating the correct one for the inbreeding coefficient computation?

**Comment by jigold** [comment]
jenkins

**Comment by jigold** [comment]
jenkins

**Comment by jigold** [comment]
@cseed : Back to you

**Pull request merged**